### ISSUE TITLE
StatusCode.OK is not respected as per the OpenTelemetry spec

### ISSUE THREAD

**Issue by apederson94** [open-telemetry/opentelemetry-java-instrumentation#10438]
### Describe the bug

When using `span.setStatus(StatusCode.OK)` from the Java SDK, the automatic instrumentation does not respect that this is final. The spec clearly states this [here](https://github.com/open-telemetry/opentelemetry-specification/blob/fa5eca7aa9d1fc5e11d09e9aec1e67aa2c0afb1f/specification/trace/api.md?plain=1#L605). This is affecting our ability to correctly administer our service and is a big problem for us.

As far as I can really tell, the issue is that there is some way that the status code is being set to ERROR after we return from our code. I have tried setting to StatusCode.OK right before we return a 500 to the caller and even that does not work. I suspect that there are however multiple locations that do not respect the spec and will need to be updated given the absoluteness of what the spec indicates.

### Steps to reproduce

1. Setup a project that uses the javaagent as instrumentation
2. During a call set the status to StatusCode.OK
3. Return a 500 from this project to the client
4. Observe in Zipkin or other span viewer that the error status is not respected and is set to StatusCode.ERROR

### Expected behavior

The span should not be marked as an error span if it has been manually set to StatusCode.OK

### Actual behavior

The span is still being set as an error span

### Javaagent or library instrumentation version

1.32.1

### Environment

**JDK**: openjdk 11.0.21
**OS**: linux

### Additional context

I am willing to sync up with anyone that will tackle this issue in a call or provide more context as is needed asynchronously. I just want to get this fixed since it is blocking us.

**Comment by trask** [comment]
> The spec clearly states this [here](https://github.com/open-telemetry/opentelemetry-specification/blob/fa5eca7aa9d1fc5e11d09e9aec1e67aa2c0afb1f/specification/trace/api.md?plain=1#L605).

@jack-berg do you think this is a bug in [`SdkSpan`](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java#L389-L401)?

**Comment by jack-berg** [comment]
Yup this seems like a pretty clear bug. Transferring to `opentelemetry-java` so we can address it in the appropriate place.
